test/test_pooling.py

```diff
@@ -411,6 +411,31 @@ def find_one():
         # maxConnecting = unbounded: 30+ connections in ~0.140+ seconds
         print(len(pool.conns))
 
+    def test_csot_timeout_message(self):
+        client = rs_or_single_client(timeoutMS=1)
+
+        with self.assertRaises(Exception) as error:
+            client.db.t.find_one({"$where": delay(2)})
+            self.assertTrue(
+                "(configured timeouts: operationTimeoutMS: 1.0ms)" in str(error.exception)
+            )
+
+    def test_socket_timeout_message(self):
+        client = rs_or_single_client(socketTimeoutMS=1)
+
+        with self.assertRaises(Exception) as error:
+            client.db.t.find_one({"$where": delay(2)})
+            self.assertTrue("(configured timeouts: socketTimeoutMS: 1.0ms)" in str(error.exception))
+
+    def test_connection_timeout_message(self):
+        client = rs_or_single_client("badhost", connectTimeoutMS=1, serverSelectionTimeoutMS=10)
```

Comment:
Might need to use a failpoint to fail connection creation, otherwise this will result in a ServerSelectionTimeoutError, not a pool checkout error.

Reply:
Wouldn't failing connection creation result in a `ServerSelectionTimeoutError`? In this case we want to have the client successfully select the server and then timeout when it tries to execute a command, correct?